osbrain/tests/test_proxy.py

```diff
@@ -79,6 +79,21 @@ def test_wrong_nameserver_address():
         locate_ns('127.0.0.1:22', timeout=1.)
 
 
+def test_no_timeout_locate_ns_not_existing():
+    """
+    Locating a NS that does not exist with no timeout should raise an error.
+    """
+    with pytest.raises(TimeoutError):
+        locate_ns('127.0.0.1:22', timeout=0.)
```

Comment:
Or we could even try `assert ... < 0.5` for that case and then add another assertion for `test_wrong_nameserver_address`: `assert 0.5 < ... < 1.5`.